server/jetstream_benchmark_test.go

```diff
@@ -1174,3 +1174,133 @@ func BenchmarkJetStreamKV(b *testing.B) {
+	)
+
+	benchmarksCases := []struct {
+		clusterSize int
+		replicas    int
+		numKeys     int
+		minBytes    int
+		maxBytes    int
+		rwRatio     float64
+	}{
+		{3, 3, 1000, 1024, 102400, 1.0},    // 3 nodes, 1000 keys, 10KB-100KB values, reads only
+		{3, 3, 1000, 1024, 102400, 0.0},    // 3 nodes, 1000 keys, 10KB-100KB values, writes only
+		{3, 3, 1000, 1024, 102400, 0.5},    // 3 nodes, 1000 keys, 10KB-100KB values, 50/50 read/writes
+		{3, 3, 1000, 1024, 102400, 0.8},    // 3 nodes, 1000 keys, 10KB-100KB values, 80/20 read/writes
+		{3, 3, 1000, 102400, 1048576, 1.0}, // 3 nodes, 1000 keys, 100KB-1MB values, reads only
+		{3, 3, 1000, 102400, 1048576, 0.0}, // 3 nodes, 1000 keys, 100KB-1MB values, writes only
+		{3, 3, 1000, 102400, 1048576, 0.5}, // 3 nodes, 1000 keys, 100KB-1MB values, 50/50 read/writes
+		{3, 3, 1000, 102400, 1048576, 0.8}, // 3 nodes, 1000 keys, 100KB-1MB values, 80/20 read/writes
+	}
+
+	for _, bc := range benchmarksCases {
+		bName := fmt.Sprintf(
+			"N=%d,R=%d,K=%d,minBytes=%db,maxBytes=%db,wrRatio=%.2f",
+			bc.clusterSize,
+			bc.replicas,
+			bc.numKeys,
+			bc.minBytes,
+			bc.maxBytes,
+			bc.rwRatio,
+		)
+		b.Run(
+			bName,
+			func(b *testing.B) {
+				if verbose {
+					b.Logf("Setting up %d nodes", bc.clusterSize)
+				}
+				var connectURL string
+				if bc.clusterSize == 1 {
+					s := RunBasicJetStreamServer(b)
+					defer s.Shutdown()
+					connectURL = s.ClientURL()
+				} else {
+					cl := createJetStreamClusterExplicit(b, "BENCH_OBJ_STORE", bc.clusterSize)
+					defer cl.shutdown()
+					cl.waitOnClusterReadyWithNumPeers(bc.clusterSize)
+					cl.waitOnLeader()
+					connectURL = cl.randomServer().ClientURL()
+				}
+
+				nc, js := jsClientConnectURL(b, connectURL)
+				defer nc.Close()
+
+				// Pre-generate all keys
+				keys := make([]string, 0, bc.numKeys)
+				for i := 1; i <= bc.numKeys; i++ {
+					key := fmt.Sprintf("%s%d", keyPrefix, i)
+					keys = append(keys, key)
+				}
+
+				// Initialize object store
+				objStoreName := fmt.Sprintf("%sTEST", objStoreNamePrefix)
+				if verbose {
+					b.Logf("Creating ObjectStore %s with R=%d", objStoreName, bc.replicas)
+				}
+				objStoreConfig := &nats.ObjectStoreConfig{
+					Bucket:   objStoreName,
+					Replicas: bc.replicas,
+				}
+				objStore, err := js.CreateObjectStore(objStoreConfig)
+				if err != nil {
+					b.Fatalf("Error creating ObjectStore: %v", err)
+				}
+
+				// Initialize keys
+				rng := rand.New(rand.NewSource(int64(seed)))
+				dataSz := rng.Intn(bc.maxBytes-bc.minBytes) + bc.minBytes
+				value := make([]byte, dataSz)
+				for _, key := range keys {
+					rng.Read(value)
+					_, err := objStore.PutBytes(key, value)
+					if err != nil {
+						b.Fatalf("Failed to initialize %s/%s: %v", objStoreName, key, err)
+					}
+				}
+
+				b.ResetTimer()
+				b.SetBytes(int64((bc.minBytes + bc.maxBytes) / 2))
+
+				var errors int
+
+				for i := 1; i <= b.N; i++ {
+					rwOpDecider := float64(rng.Int())
```

Comment:
I don't see how this can possibly work.
https://pkg.go.dev/math/rand#Rand.Float64